device/device.go

```diff
@@ -4,137 +4,107 @@ import (
+		"Application UUID: %d, "+
 			"Board type: %s, "+
+			"Name: %s, "+
 			"Local UUID: %s, "+
 			"Resin UUID: %s, "+
-			"Application UUID: %d, "+
-			"Application name: %s, "+
 			"Commit: %s, "+
 			"Target commit: %s, "+
 			"Status: %s, "+
-			"Progress: %2.2f, "+
 			"Config: %v, "+
 			"Target config: %v, "+
 			"Environment: %v, "+
 			"Target environment: %v",
-		d.Name,
+			"Restart: %t, "+
+			"Delete: %t",
+		d.ApplicationUUID,
 		d.BoardType,
+		d.Name,
 		d.LocalUUID,
 		d.ResinUUID,
-		d.ApplicationUUID,
-		d.ApplicationName,
 		d.Commit,
 		d.TargetCommit,
 		d.Status,
-		d.Progress,
 		d.Config,
 		d.TargetConfig,
 		d.Environment,
-		d.TargetEnvironment)
+		d.TargetEnvironment,
+		d.RestartFlag,
+		d.DeleteFlag)
 }
 
-func Create(boardType board.Type, name, localUUID, resinUUID string, applicationUUID int, applicationName, targetCommit string) (*Device, error) {
-	log := hook.Create(resinUUID)
-
-	board, err := board.Create(boardType, localUUID, log)
-	if err != nil {
-		return nil, err
-	}
-
-	d := &Device{
-		Log:             log,
-		Board:           board,
-		Name:            name,
+func New(applicationUUID int, boardType board.Type, name, localUUID, resinUUID string) Device {
+	return Device{
+		ApplicationUUID: applicationUUID,
 		BoardType:       boardType,
+		Name:            name,
 		LocalUUID:       localUUID,
 		ResinUUID:       resinUUID,
-		ApplicationUUID: applicationUUID,
-		ApplicationName: applicationName,
 		Commit:          "",
-		TargetCommit:    targetCommit,
+		TargetCommit:    "",
 		Status:          status.OFFLINE,
-		Progress:        0.0,
-	}
-
-	return d, d.putDevice()
-}
-
-// Only set the is_online field if the device is_online state has changed
-func (d *Device) SetStatus(newStatus status.Status) []error {
-	// oldOnline := true
-	// if d.Status == status.OFFLINE {
-	//     oldOnline = false
-	// }
-
-	d.Status = newStatus
-
-	newOnline := true
-	if d.Status == status.OFFLINE {
-		newOnline = false
 	}
-
-	// Send is_online if the status has changed or its the first time after a restart
-	// if oldOnline != newOnline || !d.statusFlag {
-	//     d.statusFlag = true
-	return supervisor.DependentDeviceInfoUpdateWithOnlineState(d.ResinUUID, (string)(d.Status), d.Commit, newOnline)
-	// }
-	// return supervisor.DependentDeviceInfoUpdateWithoutOnlineState(d.ResinUUID, (string)(d.Status), d.Commit)
-}
-
-func (d *Device) Marshall() ([]byte, error) {
-	return json.Marshal(d)
 }
 
-func Unmarshall(bytes []byte) (*Device, error) {
-	d, err := unmarshall(bytes)
-	if err != nil {
-		return nil, err
-	}
-
+func (d *Device) PopulateBoard() error {
 	log := hook.Create(d.ResinUUID)
 
-	board, err := board.Create(d.BoardType, d.LocalUUID, log)
-	if err != nil {
-		return nil, err
+	switch d.BoardType {
+	case board.MICROBIT:
+		d.Board = microbit.Microbit{
+			Log: log,
+			Micro: nrf51822.Nrf51822{
+				Log:                 log,
+				LocalUUID:           d.LocalUUID,
+				Firmware:            nrf51822.FIRMWARE{},
```

Comment:
Nope, because we have 2 `board` types: `microbit` and `nrf51822dk` and these both use the same `micro` internally: `nrf51822`